### ISSUE TITLE
[Example Request]: Lasso Tool

### ISSUE THREAD

**Issue by tobias0409** [tldraw/tldraw#4847]
### What's the example?

The lasso tool in drawing software allows users to make freeform, precise selections by manually drawing around an object, ideal for isolating irregular shapes and intricate details.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Zenkylo** [comment]
Agreed a lasso tool would be helpful. In addition or similar to the lasso, the ability to not only make free-form selections but also  the ability to select circles, squares, etc. I believe this is called the _marquee_ tool in Photoshop. It allows users to make an absolute selection, then take actions on the selection. For example, dragging or deleting the contents withing the selection.
![image](https://github.com/user-attachments/assets/f1587cd5-6703-4b03-82e2-7ac0884af44e)

My particular use case that brought me to this lasso request was needing the ability to crop the inner part of an image. Today, the crop tools allows users to crop an image by dragging along the edges. 

If I wanted to do the following to the above image, I could not.
<img width="700" alt="image" src="https://github.com/user-attachments/assets/f7c2914d-1032-42c4-a54f-8c4cf4996cce">

**Comment by SomeHats** [comment]
tldraw is primarily an _object-based_ editor. These types of interactions (selecting and manipulating within an object) are much more common / practical in _pixel-based_ editors. Think about the difference between something like figma or illustrator or miro and photoshot. Tldraw is much more about whiteboarding than it is about image editing. A really good example of how this limitation plays out is our eraser tool - it will erase the entire _object_ that it touches, not just the part of it you erase over. We're unlikely to see pixel-level selection/manipulation added to tldraw any time soon.

Something like an object-level marquee tool could be interesting though! Our of interest, do you know about "scribble select"? it's a bit similar, and you can get it in tldraw by holding the alt key. Does that fit your use-case at all?

![Image](https://github.com/user-attachments/assets/87b6b899-2131-4fb1-a071-29d1b6126826)
